warp/yul/ToCairoVisitor.py

```diff
@@ -118,88 +120,29 @@ def visit_literal(self, node: ast.Literal) -> str:
-            return f"u256_add({args_repr})"
-        elif "checked_sub" in function_name:
-            return f"uint256_sub({args_repr})"
-        elif "revert" in function_name:
-            return f"assert 0 = 1"
-        else:
-            return f"{function_name}({args_repr})"
-
     def visit_assignment(self, node: ast.Assignment) -> str:
-        self.preamble = False
-        value_repr: str = self.print(node.value)
-        if isinstance(node.value, ast.FunctionCall):
-            function_name = node.value.function_name.name
-            function_args: str = ", ".join(self.print(x) for x in node.value.arguments)
-            ids_repr: str = self.generate_ids_typed(node.variable_names, function_name)
-            self.preamble = True
-            if function_name in YUL_BUILTINS_MAP.keys():
-                builtin_to_cairo = YUL_BUILTINS_MAP[function_name](function_args)
-                merge_imports(self.imports, builtin_to_cairo.required_imports())
-                if not node.variable_names:
-                    return builtin_to_cairo.generated_cairo
-                else:
-                    return f"""{builtin_to_cairo.preamble}
-let ({ids_repr}) = {builtin_to_cairo.function_call}
-{builtin_to_cairo.ref_copy}"""
-            else:
-                call = self.remove_checked(function_args, function_name)
-                if not node.variable_names:
-                    return value_repr
-                else:
-                    ids_repr: str = ", ".join(
-                        self.print(x) for x in node.variable_names
-                    )
-                    if function_name in self.public_functions:
-                        ids_repr_felt = self.to_high_low_felt(ids_repr)
-                        return f"let ({ids_repr_felt}) = {call}\n" + self.init_u256(
-                            ids_repr
-                        )
-                    else:
-                        return f"let ({ids_repr}) = {call}\n"
-
-        else:
-            ids_repr: str = self.generate_ids_typed(node.variable_names)
-            self.preamble = True
-            return f"{ids_repr} = {value_repr}"
+        return self.visit(
+            ast.VariableDeclaration(
+                variables=[ast.TypedName(x.name) for x in node.variable_names],
+                value=node.value,
+            )
+        )
 
     def visit_function_call(self, node: ast.FunctionCall) -> str:
         fun_repr = self.print(node.function_name)
         args_repr = ", ".join(self.print(x) for x in node.arguments)
-        if fun_repr in self.public_functions:
-            args_repr = self.to_high_low_felt(args_repr)
         if fun_repr == "revert":
-            return "assert 0 = 1"
+            return "assert 0 = 1\njmp rel 0"
         elif fun_repr.startswith("checked_add"):
             merge_imports(self.imports, {"evm.uint256": {"u256_add"}})
             return f"u256_add({args_repr})"
-        elif fun_repr.startswith("checked_sub"):
```

Comment:
Why are we no longer checking for checked_sub?

Reply:
`checked_sub` doesn't act like `uint256_sub` sometimes. Sometimes, it takes only one argument and subtracts some hard-coded number from it. I suspect that it's also the case with `checked_add` in some cases but we'll see.